DataFormats/EcalRecHit/interface/EcalRecHitComparison.h

```diff
@@ -15,5 +15,10 @@ inline bool operator<( const EcalRecHit& one, const uint32_t& detid) {
 inline bool operator<( const uint32_t& detid, const EcalRecHit& other) { 
   return detid < other.detid();}
 
+// this function is meant to be to be used with std::sort and alike
```

Comment:
It would be better to use a lambda expression. Please, see an example here https://github.com/cms-sw/cmssw/pull/24549/files#r217999589